checker/dns/txt/txt.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 The Wait4X Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package txt
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"regexp"
+
+	"wait4x.dev/v2/checker"
+)
+
+// Option configures an DNS TXT records
+type Option func(d *TXT)
+
+// TXT represents DNS TXT data structure
+type TXT struct {
+	nameserver     string
+	address        string
+	expectedValues []string
+	resolver       *net.Resolver
+}
+
+// New creates the DNS TXT checker
+func New(address string, opts ...Option) checker.Checker {
+	d := &TXT{
+		address: address,
+	}
+
+	// apply the list of options to TXT
+	for _, opt := range opts {
+		opt(d)
+	}
+
+	// Nameserver settings.
+	d.resolver = net.DefaultResolver
```

Comment:
I believe you can specify the default resolver during struct initialization.
```suggestion
	d := &TXT{
		address: address,
		resolver: net.DefaultResolver,
	}

	// apply the list of options to TXT
	for _, opt := range opts {
		opt(d)
	}
```